templates/triton_services/pytorch_app.py

```diff
@@ -0,0 +1,56 @@
+"""
+Some of those code are taken from Hugging Face's documentation and modified to run on
+Ray Serve.
+
+For Hugging Face's original doc see:
+    https://huggingface.co/docs/diffusers/en/optimization/torch2.0#torchcompile
+"""
+
+from io import BytesIO
+from fastapi import FastAPI
+import torch
+
+from ray import serve
+from diffusers import DiffusionPipeline
+from fastapi.responses import StreamingResponse
+
+app = FastAPI()
+
+
+@serve.deployment(
+    ray_actor_options={
+        "num_gpus": 1,
+        "num_cpus": 1,
+    },
+)
+@serve.ingress(app)
+class PytorchDeployment:
+    def __init__(self, run_compile: bool = False):
+        self.pipe = DiffusionPipeline.from_pretrained(
+            "runwayml/stable-diffusion-v1-5",
+            torch_dtype=torch.float16,
+            use_safetensors=True,
+        ).to("cuda")
+        self.pipe.unet.to(memory_format=torch.channels_last)
+
+        if run_compile:
+            self.pipe.unet = torch.compile(
+                self.pipe.unet, mode="reduce-overhead", fullgraph=True,
+            )
+            # This will compile the model to reduce overhead and improve performance.
+            _ = self.pipe("test").images[0]
```

Comment:
I believe torch.compile can add overhead for multiple calls. would suggest that your benchmark throws out the first N requests for each condition

Reply:
Yea, from my test the very first call is super slow (took ~ 1 minute) That's why I added this to the constructor so the any real call from the benchmark are fast 😄